lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -0,0 +1,280 @@
+            device_payload['vlan'] = vlan
+            device_payload['deployment'] = module.params['deployment']
+            device_payload['extensionValues'] = ""
+            device_payload['instanceValues'] = ""
+            device_payload['freeformConfig'] = ""
+
+        payload['lanAttachList'].append(device_payload)
+
+    payload_list.append(payload)
+    json_data = json.dumps(payload_list)
+
+    return json_data
+
+
+def vrf_deploy_payload(module):
+
+    payload = dict()
+    payload['vrfNames'] = module.params['vrf_name']
+
+    json_data = json.dumps(payload)
+
+    return json_data
+
+
+def main():
+    """ main entry point for module execution
+    """
+
+    element_spec = dict(
+        action=dict(required=True, choices=['create', 'attach', 'deploy']),
+        fabric=dict(required=True, type='str'),
+        vrf_name=dict(required=True, type='str'),
+        vrf_template=dict(default='Default_VRF_Universal', type='str'),
+        vrf_extension_template=dict(default='Default_VRF_Extension_Universal', type='str'),
+        vrf_template_config=dict(default='', type='str'),
+        vrf_id=dict(type=int),
+        serial_numbers_vlans=dict(type='list'),
+        deployment=dict(default=False, type='bool')
+    )
+
+    required_one_of = [['vrf_id', 'vrf_name']]
+    mutually_exclusive = [['vrf_id'],
+                          ['serial_numbers_vlans']]
+
+    module = AnsibleModule(argument_spec=element_spec,
+                           required_one_of=required_one_of,
+                           mutually_exclusive=mutually_exclusive,
+                           supports_check_mode=True)
+
+    result = dict(
+        changed=False,
+        response=dict()
+    )
+
+    action = module.params['action']
+    fabric = module.params['fabric']
+
+    method = 'POST'
+    path = '/rest/top-down/fabrics/{}/vrfs'.format(fabric)
+
+    conn = Connection(module._socket_path)
+
+    if action == 'create':
+        if not check_vrf_exists(module, conn):
+            json_data = vrf_create_payload(module)
+            result['response'] = conn.send_request(method, path, json_data)
+            result['changed'] = True
+        else:
+            result['changed'] = False
+    elif action == 'attach':
+        json_data = vrf_attach_payload(module)
+        path += '/attachments'
+        result['response'] = conn.send_request(method, path, json_data)
+        result['changed'] = True
+    else:
+        json_data = vrf_deploy_payload(module)
+        path += '/deployments'
+        result['response'] = conn.send_request(method, path, json_data)
+        result['changed'] = True
+
+    if isinstance(result['response'], list):
+        if result['response']:
+            resp = list()
+            if result['response'][0].get('ERROR'):
+                resp.append(result['response'][0])
+                if action == 'create':
+                    resp.append("VRF ID is already in use")
+                module.fail_json(msg=resp)
```

Comment:
Nit: This block feels a little noisy. Below is more concise but it's a matter of taste I guess:

```
    res = result['response']
    if res and isinstance(res, list) and res[0].get('ERROR'):
        vrf_dupe = "VRF ID is already in use" if (action == 'create') else ''
        module.fail_json(msg=[res[0], vrf_dupe])
```

Again, your call. The rest of your code looks good. Merge away. 👍

Reply:
This is good, I will update the if conditions on both PRs before merging. Thanks.